lib/makara/config_parser.rb

```diff
@@ -158,29 +169,45 @@ def id
       end
     end
 
-    def master_configs
+    def primary_configs
       all_configs
-        .select { |config| config[:role] == 'master' }
+        .select { |config| config[:role] == 'primary' }
         .map { |config| config.except(:role) }
     end
 
+    def master_configs
+      warn "#{self.class}#master_configs is deprecated. Switch to #primary_configs"
+      primary_configs
+    end
+
     def replica_configs
       all_configs
-        .reject { |config| config[:role] == 'master' }
+        .reject { |config| config[:role] == 'primary' }
         .map { |config| config.except(:role) }
     end
 
     def slave_configs
-      warn "#slave_configs is deprecated. Switch to #replica_configs"
+      warn "#{self.class}#slave_configs is deprecated. Switch to #replica_configs"
       replica_configs
     end
 
     protected
 
     def all_configs
-      @makara_config[:connections].map do |connection|
-        base_config.merge(makara_config.except(:connections))
-                   .merge(connection.symbolize_keys)
+      @all_configs ||= @makara_config[:connections].map do |connection|
```

Comment:
This seems like a behavior change. Now we won't be re-calculating all_configs each time it is called. That may ultimately be desired behavior but wanted to call out that potentially this could break things

Reply:
AFAIK these are immutable structures anyway (since they're just loaded from config files)